services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -109,9 +111,28 @@ class HakemusService(
                     hanke = hanke))
         val hakemus = entity.toHakemus()
         hakemusLoggingService.logCreate(hakemus, userId)
+        // A hanke with a hakemus should be in RAKENTAMINEN phase
+        updateHankevaiheToRakentaminen(hanke, userId)
         return hakemus
     }
 
+    /** Update the hankevaihe of a Hanke to RAKENTAMINEN if it's not already there. */
+    private fun updateHankevaiheToRakentaminen(hanke: HankeEntity, userId: String) {
+        if (hanke.vaihe != Hankevaihe.RAKENTAMINEN) {
+            val geometriatMap = hankealueService.geometryMapFrom(hanke.alueet)
+            val hankeBeforeUpdate = HankeMapper.domainFrom(hanke, geometriatMap)
+            hanke.vaihe = Hankevaihe.RAKENTAMINEN
+            hanke.version = hanke.version?.inc() ?: 1
+            hanke.modifiedByUserId = userId
+            hanke.modifiedAt = getCurrentTimeUTCAsLocalTime()
+            hanke.generated = false
```

Comment:
Why is the generated tag updated here?

Reply:
This comes from HankeService.updateHanke which I took as a rererence for these settings. There, when hanke is updated, the generated flag is set to false. I thought that it should be done here as well.

Reply:
There, the update is a deliberate action taken by the user so the hanke should no longer be considered an automatically generated one. Here, the update is a side-effect of creating an application.

Creating an application for a generated hanke is effectively prohibited in the frontend, so updating it is redundant. If at some point we allow creating a second cable report application using the same generated hanke, we want the hanke to remain as a generated one. So there's no reason to update the tag here.

Reply:
Thank you for the clarification! I removed setting the flag here.